cmake/luajit.cmake

```diff
@@ -22,7 +22,7 @@ ExternalProject_Add(luajit
   EXCLUDE_FROM_ALL TRUE
   SOURCE_DIR ${LUAJIT_SRC}
   CONFIGURE_COMMAND ./configure
-  BUILD_COMMAND $(MAKE) CC=${CMAKE_C_COMPILER} CFLAGS=${CFLAGS} BUILD_MODE=static "XCFLAGS=-fPIC"
+  BUILD_COMMAND $(MAKE) CROSS=${CROSS_PREFIX} CFLAGS=${CFLAGS} BUILD_MODE=static "XCFLAGS=-fPIC"
```

Comment:
what is the CROSS_PREFIX variable set to on non arm64 targets?

Reply:
https://github.com/fluent/fluent-bit/blob/92e5e9f33f011da1ec94737dce4b31954556f165/lib/luajit-2.1.0-1e66d0f/doc/install.html#L377

It is a prefix for the path to the toolchain. It is empty for X86 builds (if the compilation is done on an X86 host machine), and luajit will just use "" + default_path for the toolchain. For other architectures, we need dedicated cmake files for each architecture to set this accordingly.